pkg/ipamd/ipamd_test.go

```diff
@@ -459,31 +460,39 @@ func getDummyENIMetadataWithV6Prefix() awsutils.ENIMetadata {
 
 func TestIncreaseIPPoolDefault(t *testing.T) {
 	_ = os.Unsetenv(envCustomNetworkCfg)
-	testIncreaseIPPool(t, false)
+	testIncreaseIPPool(t, false, false)
 }
 
 func TestIncreaseIPPoolCustomENI(t *testing.T) {
 	_ = os.Setenv(envCustomNetworkCfg, "true")
 	_ = os.Setenv("MY_NODE_NAME", myNodeName)
-	testIncreaseIPPool(t, true)
+	testIncreaseIPPool(t, true, false)
 }
 
-func testIncreaseIPPool(t *testing.T, useENIConfig bool) {
+func TestIncreaseIPPoolCustomENIOnNonSchedulableNode(t *testing.T) {
```

Comment:
nit: can you just add a comment above this test explaining that it covers that ENI is allocated even when node is unschedulable when env var is set?

And can we also cover the negative case, i.e. ENI is not allocated when node is unschedulable (and env var is not set)?

Reply:
Done